### PR TITLE
Add clientset tests for graceful deletion

### PR DESCRIPTION
**Pull request by pmorie**
Discovered during some hacking tonight - left binding for you to do.

cc @arschles @MHBauer @jpeeler

### CODE REVIEW

**Comment by MHBauer** [comment]
wrong repo for the PR?

**Comment by MHBauer** [comment]
Oh, branch of a branch. Nevermind.

File: pkg/registry/servicecatalog/binding/strategy.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func (bindingRESTStrategy) PrepareForCreate(ctx kapi.Context, obj runtime.Object
 	// Fill in the first entry set to "creating"?
 	binding.Status.Conditions = []sc.BindingCondition{}
 
-	binding.Finalizers = []string{"BindingFinalizer"}
+	binding.Finalizers = []string{"kubernetes"}
```
Comment (krancour) [inline pkg/registry/servicecatalog/binding/strategy.go]: This, I don't understand. Isn't the idea behind finalizers that there's _something_ that clears _each_ one? i.e. There could be _multiple_ somethings. "kubernetes" looks too generic and as such that I don't think we could trust that _something else_ doesn't clear this finalizer.

File: pkg/registry/servicecatalog/broker/storage.go [hunk 2]
```diff
@@ -98,9 +98,10 @@ func NewStorage(opts generic.RESTOptions) (brokers, brokersStatus rest.Storage)
 		// QualifiedResource should always be plural
 		QualifiedResource: api.Resource("brokers"),
 
-		CreateStrategy: brokerRESTStrategies,
-		UpdateStrategy: brokerRESTStrategies,
-		DeleteStrategy: brokerRESTStrategies,
+		CreateStrategy:          brokerRESTStrategies,
+		UpdateStrategy:          brokerRESTStrategies,
+		DeleteStrategy:          brokerRESTStrategies,
+		EnableGarbageCollection: true,
```
Comment (krancour) [inline pkg/registry/servicecatalog/broker/storage.go]: Is this addition the key to making the apiserver complete a real delete after it receives an update with an empty finalizers list?

File: pkg/registry/servicecatalog/binding/strategy.go (hunk 1 above)
Comment (pmorie) [inline pkg/registry/servicecatalog/binding/strategy.go]: So, it looks like in our vendored version of kube, the only valid finalizers are 'kubernetes' and 'orphan'.  So, use 'kubernetes' for now.

File: pkg/registry/servicecatalog/broker/storage.go (hunk 2 above)
Comment (pmorie) [inline pkg/registry/servicecatalog/broker/storage.go]: Yep

File: pkg/registry/servicecatalog/binding/strategy.go (hunk 1 above)
Comment (krancour) [inline pkg/registry/servicecatalog/binding/strategy.go]: What makes only those two valid? i.e. How is that enforced?

File: pkg/registry/servicecatalog/broker/storage.go (hunk 2 above)
Comment (krancour) [inline pkg/registry/servicecatalog/broker/storage.go]: Cool.

**Pull request merged**